pkg/controller/master-controller-manager/user-synchronizer/controller.go

```diff
@@ -84,39 +82,14 @@ func Add(
 	}
 
 	if err := c.Watch(
-		&source.Kind{Type: &kubermaticv1.User{}}, &handler.EnqueueRequestForObject{}, serviceAccountPredicate, withEventFilter(),
+		&source.Kind{Type: &kubermaticv1.User{}}, &handler.EnqueueRequestForObject{}, serviceAccountPredicate,
 	); err != nil {
 		return fmt.Errorf("failed to create watch for user objects in master cluster: %w", err)
 	}
 
 	return nil
 }
 
-func withEventFilter() predicate.Predicate {
-	return predicate.Funcs{
-		CreateFunc: func(e event.CreateEvent) bool {
-			return true
-		},
-		UpdateFunc: func(e event.UpdateEvent) bool {
-			oldUser, ok := e.ObjectOld.(*kubermaticv1.User)
-			if !ok {
-				return false
-			}
-			newUser, ok := e.ObjectNew.(*kubermaticv1.User)
-			if !ok {
-				return false
-			}
-			return !reflect.DeepEqual(oldUser.Spec, newUser.Spec)
```

Comment:
I am having a problem to understand how removing update predicate fixes the issue. Correct me if I am wrong, but if cluster admin creates a new user this user is being synchronized to all seeds, correct? So the communication is from master to seed for synchronization. If cluster admins delete a user then we should get a delete event from delete function. So how the update function was problematic?

Reply:
The issue is fixed because the deleting a resource with a finalizer actually does an update with deletionTimestamp, we ignore that in the event filter because we just check spec, so it doesnt get handled.

Good idea with comparing the generations, that should work. @imharshita could you change it?